cmd/monitor/monitor.go

```diff
@@ -36,6 +36,31 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 )
 
+var (
+	lastTestDuration = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "test_duration_seconds",
+		Help: "Duration of the last completed test (seconds)",
+	})
```

Comment:
This is probably only good as long as durations exceed the Prometheus scrape interval. @gabemontero can you refresh my memory on prior discussion about this? I could have sworn you said there was some support for a way to allow these to accumulate locally.

Reply:
hey @ironcladlou 

Yeah, I mentioned something along  those lines.  Specifically, if you implement the collector interface yourself, you can control how many metrics are returned to prometheus.  Your collector could then accumulate metrics how it sees fit.

Most likely, you'd want to leverage a constant metric as well (our builds metrics and my jenkins smoke test do this), but I have not experimented with passing in other types ... there may be flexibility there.

See https://github.com/gabemontero/monitor-jenkins-lifecycle/blob/master/cmd/monitor/monitor.go and https://github.com/openshift/origin/blob/master/pkg/build/metrics/prometheus/metrics.go

The are not maintaining the collection of metrics right now, but I've got some comments in https://github.com/gabemontero/monitor-jenkins-lifecycle/blob/master/cmd/monitor/monitor.go about where I would switch from a singleton to a collection, and have tried it successfully, though did not commit the change.